File: backend/src/main/java/com/healthsphere/backend/service/PatientService.java
package com.healthsphere.backend.service;

import com.healthsphere.backend.dto.PatientRequest;
import com.healthsphere.backend.dto.PatientResponse;

import java.util.List;

public interface PatientService {

    PatientResponse createPatient(PatientRequest request);

    List<PatientResponse> getAllPatients();

    PatientResponse getPatientById(Long id);

    PatientResponse updatePatient(Long id, PatientRequest request);

    void deletePatient(Long id);
}

File: frontend/src/services/patientService.js
import api from "./api";

const URL = "/api/patients";

const patientService = {

    getAllPatients() {
        return api.get(URL);
    },

    getPatientById(id) {
        return api.get(`${URL}/${id}`);
    },

    createPatient(patient) {
        return api.post(URL, patient);
    },

    updatePatient(id, patient) {
        return api.put(`${URL}/${id}`, patient);
    },

    deletePatient(id) {
        return api.delete(`${URL}/${id}`);
    }
};

export default patientService;